autonomy/chain/base.py

```diff
@@ -0,0 +1,110 @@
+# -*- coding: utf-8 -*-
+# ------------------------------------------------------------------------------
+#
+#   Copyright 2022 Valory AG
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+
+"""Chain interaction base classes."""
+
+import json
+from enum import Enum
+from typing import Any, Dict, Iterable, List, Optional
+
+from aea.crypto.base import Crypto, LedgerApi
+
+from autonomy.chain.config import ContractConfig
+from autonomy.data import DATA_DIR
+
+
+ABI_DIR = DATA_DIR / "abis"
+
+ContractInterfaceType = Any
+
+
+def get_abi(filename: str) -> Dict:
```

Comment:
We already have this elsewhere I’m sure as we already had the service registry

Reply:
I know, I'm planning to move that logic from `autonomy.deploy.chain` to `autonomy.chain`

Reply:
Addressed